### ISSUE TITLE
Exception when using Lamar for DI

### ISSUE THREAD

**Issue by sreagleeye** [AMalininHere/camunda-worker-dotnet#3]
When using the IoC tool [Lamar](https://jasperfx.github.io/lamar/), I am getting an exception when trying to run a worker. The exception occurs on the following line within the `WorkerHostedService` 
```
protected override Task ExecuteAsync(CancellationToken stoppingToken)
{
    var activeTasks = Enumerable.Range(0, _options.WorkerCount)
        .Select(_ => _serviceProvider.GetRequiredService<ICamundaWorker>())
        .Select(worker => worker.Run(stoppingToken))
        .ToList();
    return Task.WhenAll(activeTasks);
}
```
The exception is as follows
```
Frame type Lamar.IoC.Resolvers.CastRootScopeFrame does not implement IResolverFrame
at Lamar.IoC.Instances.FuncResolverDefinition.BuildResolver()
at Lamar.IoC.Instances.GeneratedInstance.BuildFuncResolver(Scope scope)
at Lamar.IoC.Instances.GeneratedInstance.buildResolver(Scope scope)
at Lamar.IoC.Instances.GeneratedInstance.ToResolver(Scope topScope)
at Lamar.ServiceGraph.FindResolver(Type serviceType)
at Lamar.IoC.Scope.GetInstance(Type serviceType)
at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
at Camunda.Worker.Execution.WorkerHostedService.<>c__DisplayClass3_0.<ExecuteAsync>b__0(Int32 _) in C:\GitHub\camunda-worker-dotnet\src\Camunda.Worker\Execution\WorkerHostedService.cs:line 33
at System.Linq.Utilities.<>c__DisplayClass2_0`3.<CombineSelectors>b__0(TSource x)
at System.Linq.Enumerable.SelectIPartitionIterator`2.ToList()
at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
at Camunda.Worker.Execution.WorkerHostedService.ExecuteAsync(CancellationToken stoppingToken) in C:\GitHub\camunda-worker-dotnet\src\Camunda.Worker\Execution\WorkerHostedService.cs:line 32
at Microsoft.Extensions.Hosting.BackgroundService.StartAsync(CancellationToken cancellationToken)
at Microsoft.AspNetCore.Hosting.Internal.HostedServiceExecutor.<>c__DisplayClass3_0.<StartAsync>b__0(IHostedService service)
at Microsoft.AspNetCore.Hosting.Internal.HostedServiceExecutor.ExecuteAsync(Func`2 callback)
```

The root of the issue is that Lamar is unable to inject an `IServiceScopeFactory` into the `DefaultCamundaWorker`. I'm not 100% clear on why that is. However, replacing the `IServiceScopeFactory` with a `IServiceProvider` works as intended. This also appears to achieve the same behavior since the `CreateScope` function is the same across both an `IServiceScopeFactory` and `IServiceProvider`.

**Comment by AMalininHere** [comment]
It looks weird because [`IServiceProvider#CreateScope`](https://github.com/aspnet/Extensions/blob/689c4a891cc0783d6c7ed1f9bab1e9854981e65b/src/DependencyInjection/DI.Abstractions/src/ServiceProviderServiceExtensions.cs#L127) uses `IServiceScopeFactory` to create scopes

**Comment by sreagleeye** [comment]
Totally agreed. I saw the same code you referenced. I was considering opening an issue in Lamar as well, but when I saw that `IServiceProvider` worked as an alternative, I thought that might be the better approach.

**Comment by AMalininHere** [comment]
It's included to `0.6.0` release. Thank you.
